pymeasure/instruments/keysight/keysight34980A.py

```diff
@@ -0,0 +1,628 @@
+        by the READ? command, the FETCh? command, or other queries of scan results.
+        """
+        self.write(":FORMat:READing:UNIT OFF")
+        while((self.values(":FORMat:READing:UNIT?")) == 1):
+            self.write(":FORMat:READing:UNIT OFF")
+
+    def is_calibrated(self, sec_code = "AG34980"):
+        """
+        The following command performs a calibration on the Internal DMM and returns a pass/fail indication.
+        """
+        self.device_unsecure(sec_code)
+        temp_values = self.values(":CALibration?")
+        temp = str(temp_values[0])
+        self.device_secure(sec_code)
+        return(temp)
+
+    def calibrate_security_code_modify(self, old_sec_code = "AG34980", new_sec_code = "AG34980"):
+        """
+        This command allows you to enter a new security code to prevent accidental or 
+        unauthorized calibrations. The specified code is used to unsecure the mainframe 
+        and all installed modules.
+        """
+        self.device_unsecure(old_sec_code)
+        self.write(":CALibration:SECure:CODE ," + str(new_sec_code))    
+        self.device_secure(new_sec_code)
+
+    def device_unsecure(self, sec_code = "AG34980"):
+        """
+        This command unsecures the instrument for calibration.         
+        """
+        self.write(":CALibration:SECure:STATe OFF," + str(sec_code))
+
+    def device_secure(self, sec_code = "AG34980"):
+        """
+        This command secures the instrument for calibration. 
+        """
+        self.write(":CALibration:SECure:STATe ON," + str(sec_code))
+
+    def is_calibration_security_setting(self):
+        """
+        This command checks the calibration security setting of the instrument.         
+        """
+        temp_values = self.values(":CALibration:SECure:STATe?")
+        temp = str(temp_values[0])
+        return(temp)
+
+    def line_freq(self, freq_val = 60):
+        """
+        This command sets the power-line reference frequency used by the 
+        internal DMM's analog-to-digital converter.        
+        """
+        self.write(":CALibration:LFRequency " + str(freq_val))
+        temp = self.values(":CALibration:LFRequency?")
+        while(temp[0] != freq_val):
+                self.write(":CALibration:LFRequency " + str(freq_val))
+                temp = self.values(":CALibration:LFRequency?")
+
+    def measure_line_freq(self):
+        """
+        This command checks the power-line reference frequency used by the 
+        internal DMM's analog-to-digital converter.        
+        """
+        temp_values = self.values(":CALibration:LFRequency?")
+        temp = str(temp_values[0])    
+        return(temp)
+
+    def internal_dmm_enable(self):
+        """
+        This command enables the internal DMM. 
+        """
+        self.write(":INSTrument:DMM:STATe ON")
+        while(self.internal_dmm_state() == 0):
+            self.write(":INSTrument:DMM:STATe ON")
```

Comment:
I see you often have such a construction; could you explain why this is needed? Is setting a value once not sufficient?

Reply:
It is just a verification to check that the value is set correctly.

Reply:
Do you know if it happens often/at all that the value isn't correctly set? I'd think that it would not be required in general and that it just clutters the communication. Moreover, if there is another reason a certain value cannot be set, it will try again indefinitely.